mne/decoding/transformer.py

```diff
@@ -245,6 +245,54 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(object):
+    """Mixin class for transformation of different data to fit into
+       scikit-learn pipeline.
+    """
+
+    def fit(self, data, y=None):
+        """Fit data and parameters into self. This is defined for the sake
+           of completeness of scikit-learn API.
+        
+        Parameters
+        ----------
+        data : array, 2D or 3D matrix
+        y : None or array of labels
+
+        Returns
+        -------
+        self : instance of VectorizerMixin
+        """
+        return self
+
+    def transform(self, data, y=None):
```

Comment:
Make sure to catch methods from sklearn superclasses and raise not
implemented errors if you don't implement. Otherwise we have methods
hanging around that don't work.
On Wed, 11 May 2016 at 20:59, Mainak Jas <notifications@github.com> wrote:

> In mne/decoding/transformer.py
> <https://github.com/mne-tools/mne-python/pull/3212#discussion_r62906280>:
>
> > +    def fit(self, data, y=None):
> > +        """Fit data and parameters into self. This is defined for the sake
> > +           of completeness of scikit-learn API.
> > +
> > +        Parameters
> > +        ----------
> > +        data : array, 2D or 3D matrix
> > +        y : None or array of labels
> > +
> > +        Returns
> > +        -------
> > +        self : instance of VectorizerMixin
> > +        """
> > +        return self
> > +
> > +    def transform(self, data, y=None):
>
> also you need fit_transform ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3212/files/4e77003ad08d76febacba216027ab0b99d3fa4fa#r62906280>
>